### PR TITLE
gnrc: rename ng_netbase to gnrc

### PR DESCRIPTION
**Pull request by authmillenon**
Renames `ng_netbase` to `gnrc` and adds `gnrc` as a factual module rather than a pseudo module. The reasoning is to add the path to `gnrc`'s submodules to its Makefile (in a similar fassion to `sys/Makefile` currently) as soon as they are moved into it.

Steps I took:

```bash
git mv sys/include/net/ng_netbase.h sys/include/net/gnrc.h
git grep --name-only "ng_netbase" | xargs sed -i 's/ng_netbase/gnrc/g'
```

After that I added the `gnrc` module as an empty directory with a preliminary Makefile, added it as dependency for all `gnrc` submodules except `gnrc_netapi`, `gnrc_netif`, `gnrc_netif_hdr`, `gnrc_netreg`, and `gnrc_pktbuf` (the module `gnrc` is build upon itself) and made `sys/gnrc` the build path for `gnrc` an all its submodules in `sys/Makefile`.

### CODE REVIEW

**Comment by authmillenon** [comment]
Rebased to current master.

**Comment by authmillenon** [comment]
Rebased to current master

**Comment by OlegHahm** [comment]
Rebase required.

**Comment by authmillenon** [comment]
Rebased to current master.

**Comment by authmillenon** [comment]
Rebased to current master.

File: sys/include/net/gnrc.h [hunk 1]
```diff
@@ -7,12 +7,13 @@
  */
 
 /**
- * @defgroup    net_ng_netbase  Network Stack Pseudo Header
+ * @defgroup    net_gnrc    Generic (gnrc) network stack.
  * @ingroup     net
+ * @brief       RIOT's modular default network stack.
```
Comment (OlegHahm) [inline sys/include/net/gnrc.h]: `s/default network/default IPv6 network/`

Comment (authmillenon) [inline sys/include/net/gnrc.h]: GNRC's modularity doesn't restrict it to IPv6.

File: sys/include/net/ng_udp.h [hunk 2]
```diff
@@ -54,6 +54,17 @@ extern "C" {
...
 /**
+ * @brief   UDP header definition
+ */
+typedef struct __attribute__((packed)) {
+    network_uint16_t src_port;      /**< source port, in network byte order */
+    network_uint16_t dst_port;      /**< destination port, network byte order */
+    network_uint16_t length;        /**< payload length (including the header),
+                                     *   network byte order */
+    network_uint16_t checksum;      /**< checksum */
+} ng_udp_hdr_t;
```
Comment (OlegHahm) [inline sys/include/net/ng_udp.h]: Seems unrelated

File: sys/net/network_layer/ng_sixlowpan/frag/rbuf.c [hunk 3]
```diff
@@ -59,16 +58,12 @@ static rbuf_t *_rbuf_get(const void *src, size_t src_len,
                          const void *dst, size_t dst_len,
                          size_t size, uint16_t tag);
 
-void rbuf_add(ng_netif_hdr_t *netif_hdr, ng_pktsnip_t *pkt,
+void rbuf_add(ng_netif_hdr_t *netif_hdr, ng_sixlowpan_frag_t *frag,
```
Comment (OlegHahm) [inline sys/net/network_layer/ng_sixlowpan/frag/rbuf.c]: This and the following changes seem to be unrelated, too.

File: sys/include/net/ng_udp.h (hunk 2 above)
Comment (authmillenon) [inline sys/include/net/ng_udp.h]: Bad rebase, I would say :/

File: sys/net/network_layer/ng_sixlowpan/frag/rbuf.c (hunk 3 above)
Comment (authmillenon) [inline sys/net/network_layer/ng_sixlowpan/frag/rbuf.c]: Something might have gone wrong during rebase. Will look into it.

**Comment by authmillenon** [comment]
Fixed rebase errors and amended immediately.

File: sys/include/net/gnrc.h (hunk 1 above)
Comment (OlegHahm) [inline sys/include/net/gnrc.h]: But currently only IPv6 is implemented and I doubt that there's ever gonna be, let's say, a CCN implementation for gnrc. So, I would put at least the word IP here.

Comment (authmillenon) [inline sys/include/net/gnrc.h]: I'm still not convinced, but I added `IP` anyways.

**Comment by OlegHahm** [comment]
ACK, please squash

**Comment by authmillenon** [comment]
Squashed.
